src/app/config/or_state_st.h

```diff
@@ -87,6 +87,18 @@ struct or_state_t {
 
   /** When did we last rotate our onion key?  "0" for 'no idea'. */
   time_t LastRotatedOnionKey;
+
+  /** Do we automatically fail IPv4 or IPv6? If so, set one of these to 1. */
+  int IPv4AutoFail;
```

Comment:
We should also think about the privacy implications of storing exact times and exact failure numbers on clients.